xtask/src/commands/changeset/mod.rs

```diff
@@ -0,0 +1,499 @@
+            Classification::Maintenance,
+            Classification::Documentation,
+            Classification::Experimental,
+        ];
+        CLASSIFICATIONS.iter()
+    }
+}
+
+type ParseError = &'static str;
+impl FromStr for Classification {
+    type Err = ParseError;
+    fn from_str(classification: &str) -> Result<Self, Self::Err> {
+        match classification {
+            "break" => Ok(Classification::Breaking),
+            "feat" => Ok(Classification::Feature),
+            "fix" => Ok(Classification::Fix),
+            "config" => Ok(Classification::Configuration),
+            "maint" => Ok(Classification::Maintenance),
+            "docs" => Ok(Classification::Documentation),
+            "exp" => Ok(Classification::Experimental),
+            &_ => Err("unknown classification"),
+        }
+    }
+}
+
+impl fmt::Display for Classification {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let pretty = match self {
+            Classification::Breaking => "❗ BREAKING ❗",
+            Classification::Feature => "🚀 Features",
+            Classification::Fix => "🐛 Fixes",
+            Classification::Configuration => "📃 Configuration",
+            Classification::Maintenance => "🛠 Maintenance",
+            Classification::Documentation => "📚 Documentation",
+            Classification::Experimental => "🧪 Experimental",
+        };
+        write!(f, "{}", pretty)
+    }
+}
+
+#[derive(Debug, StructOpt)]
+pub struct Create {
+    /// Use the current branch as the file name
+    #[structopt(short = "b", long = "--with-branch-name")]
+    with_branch_name: bool,
+
+    /// The classification of the changeset
+    #[structopt(short = "c", long = "--class")]
+    classification: Option<Classification>,
+}
+
+async fn github_graphql_post_request(
+    token: &str,
+    url: &str,
+    request_body: &graphql_client::QueryBody<
+        matching_pull_request::matching_pull_request::Variables,
+    >,
+) -> Result<
+    graphql_client::Response<matching_pull_request::matching_pull_request::ResponseData>,
+    ::reqwest::Error,
+> {
+    let client = Client::builder().build()?;
+
+    let res = client
+        .post(url)
+        .header(
+            "User-Agent",
+            format!("github {} releasing", REPO_WITH_OWNER),
+        )
+        .header("Authorization", format!("Bearer {}", token))
+        .json(request_body)
+        .send()
+        .await?;
+    let response_body: graphql_client::Response<
+        matching_pull_request::matching_pull_request::ResponseData,
+    > = res.json().await?;
+    Ok(response_body)
+}
+
+impl Create {
+    pub fn run(&self) -> Result<()> {
+        let changesets_dir_path = PKG_PROJECT_ROOT.join(".changesets");
+        tokio::runtime::Builder::new_multi_thread()
+            .enable_all()
+            .build()
+            .unwrap()
+            .block_on(async {
+                let items = Classification::iterator().collect_vec();
```

Comment:
If you’re gonna collect into a `Vec` maybe it’s shouldn’t be an iterator in the first place. It could be a static slice instead. And, since it’s constant, be an associated constant instead of an associated function.

Definition:

```rust
impl Classification {
    const ALL: &'static [Self] = &[
        // …
    ];
}
```

Use:

```rust
let items = Classification::ALL;
```

Reply:
Excellent observation, suggestion and learning opportunity.  Thanks!

f35f3a82